ptp/ptp4u/server/server.go

```diff
@@ -43,6 +44,10 @@ type Server struct {
 	// server source fds
 	eFd int
 	gFd int
+
+	// drain logic
+	cancel context.CancelFunc
```

Comment:
Yeah I am afraid this against go ideology :(
https://pkg.go.dev/context

> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx:
```
func DoSomething(ctx context.Context, arg Arg) error {
	// ... use ctx ...
}
```

Reply:
That is exactly what we do from the point of view of the subscriptions. We don't store the context, we only pass it. And this context is to control the lifetime of the subscriptions.

Reply:
Well you store it in the Server struct and use it in methods like `Drain()` without passing. But I am ok if you say it's ok :)

Reply:
In the `Drain` method we cancel the context and in the `Undrain` method we create a new context. This is for managing the lifecycle of the subscriptions.

We typically have a single and unique instance of a Server. I don't see an issue with this but happy to listen to suggestions.